internal/pkg/runtime/launch/launcher_linux.go

```diff
@@ -866,6 +866,11 @@ func (l *Launcher) setNvCCLIConfig() (err error) {
 	}
 	l.engineConfig.SetNvCCLIEnv(nvCCLIEnv)
 
+	if len(l.cfg.OverlayPaths) > 0 && l.cfg.Writable {
+		sylog.Verbosef("Detected writable overlay images, skipping setting --writable-tmpfs (otherwise required by --nvccli)")
+		l.cfg.WritableTmpfs = false
+	}
+
```

Comment:
> On closer look, I don't see how this is actually doing anything. If l.cfg.Writable was already true, the next check below already shouldn't have done anything. All this is doing is turning off `--writable-tmpfs`, but it doesn't get automatically turned on until the next check. I am confused.

hmm, not fully understand this part, from my understanding, we should disable `--writable-tmpfs` when we have a writable (`l.cfg.Writable = True`) overlay path (`l.cfg.OverlayPaths > 0`)?

Reply:
Did you reproduce the problem and show that this fixed it?  I don't see how it can.  You set WritableTmpfs false here, but it isn't until the next statement that it would have been set to true, and that wouldn't have happened anyway because your new statement only happens when Writable is true, and the next statement only happens when Writable is false.

Reply:
I finally can reproduce this issue, though I can not successfully start the instance, but the `--writable-tmpfs` should be disabled. 
```
[vagrant@localhost test]$ apptainer instance run --overlay nvidia.sif --nvccli --env "TZ=UTC,DISPLAY_SIZEW=1920,DISPLAY_SIZEH=1080,DISPLAY_REFRESH=60,DISPLAY_DPI=96,DISPLAY_CDEPTH=24,PASSWD=mypasswd,SELKIES_ENCODER=nvh264enc,SELKIES_VIDEO_BITRATE=8000,SELKIES_FRAMERATE=60,SELKIES_AUDIO_BITRATE=128000,SELKIES_BASIC_AUTH_PASSWORD=mypasswd" nvidia-glx-desktop_latest.sif glx
INFO:    Setting 'NVIDIA_VISIBLE_DEVICES=all' to emulate legacy GPU binding.
INFO:    Instance stats will not be available - requires cgroups v2 with systemd as manager.
INFO:    squashfuse not found, will not be able to mount SIF or other squashfs files
INFO:    gocryptfs not found, will not be able to use gocryptfs
INFO:    Converting SIF file to temporary sandbox...
FATAL:   while extracting nvidia-glx-desktop_latest.sif: root filesystem extraction failed: extract command failed: WARNING: passwd file doesn't exist in container, not updating
WARNING: group file doesn't exist in container, not updating
WARNING: Skipping mount /etc/hosts [binds]: /etc/hosts doesn't exist in container
WARNING: Skipping mount /etc/localtime [binds]: /etc/localtime doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/tmp [tmp]: /tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/var/tmp [tmp]: /var/tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container

Write on output file failed because No space left on device

FATAL ERROR:writer: failed to write file /image/root/opt/wine-staging/lib64/wine/x86_64-windows/ntdll.dll
Parallel unsquashfs: Using 4 processors
114337 inodes (168359 blocks) to write
```

if I remove `--overlay` or add `:ro` to the overlay image
```
[vagrant@localhost test]$ apptainer instance run --overlay nvidia.sif:ro --nvccli --env "TZ=UTC,DISPLAY_SIZEW=1920,DISPLAY_SIZEH=1080,DISPLAY_REFRESH=60,DISPLAY_DPI=96,DISPLAY_CD
EPTH=24,PASSWD=mypasswd,SELKIES_ENCODER=nvh264enc,SELKIES_VIDEO_BITRATE=8000,SELKIES_FRAMERATE=60,SELKIES_AUDIO_BITRATE=128000,SELKIES_BASIC_AUTH_PASSWORD=mypasswd" nvidia-glx-de
sktop_latest.sif glx
INFO:    Setting 'NVIDIA_VISIBLE_DEVICES=all' to emulate legacy GPU binding.
INFO:    Setting --writable-tmpfs (required by nvidia-container-cli)
INFO:    Instance stats will not be available - requires cgroups v2 with systemd as manager.
INFO:    squashfuse not found, will not be able to mount SIF or other squashfs files
INFO:    gocryptfs not found, will not be able to use gocryptfs
INFO:    Converting SIF file to temporary sandbox...
^C^C

FATAL:   while extracting nvidia-glx-desktop_latest.sif: root filesystem extraction failed: extract command failed: WARNING: passwd file doesn't exist in container, not updating
WARNING: group file doesn't exist in container, not updating
WARNING: Skipping mount /etc/hosts [binds]: /etc/hosts doesn't exist in container
WARNING: Skipping mount /etc/localtime [binds]: /etc/localtime doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/tmp [tmp]: /tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/var/tmp [tmp]: /var/tmp doesn't exist in container
WARNING: Skipping mount /usr/local/var/apptainer/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container

FATAL ERROR:write_xattr: failed to write xattr user.rootlesscontainers for file /image/root/opt/wine-staging/lib64/wine/x86_64-windows/npptools.dll because no extended attribute space remaining (per file or filesystem limit)
Parallel unsquashfs: Using 4 processors
114337 inodes (168359 blocks) to write
```